### PR TITLE
Feature/sc 9738/import utilities fixes and optimization

### PR DESCRIPTION
**Pull request by vsimak**
Added a fix for sanitization of not adding item relations components when found
alone, in chunks and choice in items.ts for documents, products, and variant
items.

### CODE REVIEW

**Comment by vsimak** [comment]
I didn't actually do types for ComponentInput, I believe that it is probably done already and somewhere to be read from.
The current one was written 2 years ago, with only one input, that why I did ts ignore / swapped to any at the moment.

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 1]
```diff
@@ -1229,25 +1238,37 @@ export async function setItems({
       if (item._componentsData?.[language]) {
         Object.keys(item._componentsData[language]).forEach(
           (componentId: string) => {
-            const componentContent: ComponentContentInput =
+            const componentContent: any =
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1241]: bad bad bad. Keep the types. There are better ways of doing the bit that comes later.

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 2]
```diff
@@ -1229,25 +1238,37 @@ export async function setItems({
...
-                buildUpdateItemComponentQueryAndVariables({
-                  itemId,
-                  language,
-                  input: {
-                    componentId,
-                    ...componentContent,
-                  },
-                })
+            if (
+              // @ts-ignore
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1245]: This should be extracted to a function, that takes `componentContent: ComponentContentInput` as input. Make a descriptive name for the function, like `shouldComponentContentBeSet`, or something along those lines. That will allow us to test it in isolation too.

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 3]
```diff
@@ -1268,20 +1289,34 @@ export async function setItems({
...
-                    buildUpdateVariantComponentQueryAndVariables({
-                      productId: itemId,
-                      sku: variant.sku,
-                      language,
-                      input: {
-                        componentId,
-                        ...componentContent,
-                      },
-                    })
+                if (
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1292]: We can re-use the above proposed function here.

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 4]
```diff
@@ -1366,7 +1400,7 @@ export async function setItems({
         }),
         ...(attributes && { attributes }),
       }
-
+      delete variant.components
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1403]: Why are the variants deleted?

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 5]
```diff
@@ -1379,73 +1413,90 @@ export async function setItems({
             onUpdate,
           })
         }
+        if (jsonVariant._componentsData?.[language]) {
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1416]: Why can't this logic be contained in `createComponentsInput`?

File: src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts [hunk 6]
```diff
@@ -51,8 +51,9 @@ async function downloadFile(fileURL: string) {
   // Videos
   if (fileURL.endsWith('.m3u8')) {
     const canConvert = await ffmpegAvailable
+    console.log('canConvert: ', canConvert)
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts:54]: should be removed

File: src/bootstrap-tenant/bootstrapper/items.ts [hunk 7]
```diff
@@ -1379,73 +1413,90 @@ export async function setItems({
...
-              }
-
               return {
-                id,
-                ...(p.initial && {
-                  initial: subscriptionPlanPrincingJsonToInput(
-                    p.initial,
-                    meteredVariables
-                  ),
+                identifier: sP.identifier,
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/items.ts:1454]: This is unrelated to the components thing, right? For the future, this should be contained within its own branch and PR

File: src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts [hunk 8]
```diff
@@ -162,6 +163,7 @@ export async function remoteFileUpload({
     )
   }
 
+  console.log('creating signature', fileName, contentType)
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts:166]: remove, or wrap in a if-logging===verbose

File: src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts [hunk 9]
```diff
@@ -205,12 +208,15 @@ export async function remoteFileUpload({
...
   formData.append('file', file)
 
+  console.log('upploading the file')
   // Upload the file
   const uploadResponse = await fetch(url, {
     method: 'post',
     body: formData,
   })
 
+  console.log('upload Response', uploadResponse)
```
Comment (hakonkrogh) [inline src/bootstrap-tenant/bootstrapper/utils/remote-file-upload.ts:218]: Lots of console.logs that should be remove, or wrapped in a if-logging=verbose

**Review by hakonkrogh — CHANGES_REQUESTED** [review]

File: src/bootstrap-tenant/bootstrapper/items.ts (hunk 2 above)
Comment (vsimak) [reply src/bootstrap-tenant/bootstrapper/items.ts:1245]: I've created a hookd and extended the types, but after realized we don't need it, ive just made changes to the componentInput function where it returns null for itemRelations and then checks if componentContent is truthy before pushing it to updates array.

**Pull request merged**